File: 43주차/타겟넘버.py
answer = 0
def solution(numbers, target):
    
 
    def dfs(L,total):
        global answer
        if L == len(numbers):
            if total == target:
                answer += 1
            return
        dfs(L+1, total - numbers[L])
        dfs(L+1, total + numbers[L])
    
    dfs(0,0)
    return answer
# 프로그래머스 고득점 kit. bfs/dfs

File: 47주차/타겟넘버.js
function solution(numbers, target) {
  let answer = 0;
  const n = numbers.length;
  function dfs(L, total) {
    if (L > n) return;
    if (L === n && total === target) {
      // n개 모두 계산했을때, target과 같다면 +1.
      answer += 1;
      return;
    }
    dfs(L + 1, total + numbers[L]); // 덧셈 혹은 뺄셈.
    dfs(L + 1, total - numbers[L]);
  }
  dfs(0, 0);
  return answer;
}
// 프로그래머스 dfs.
